### DISCUSSION TITLE
Export from external button

### DISCUSSION THREAD

**Discussion by swissjust-america** [pxlrbt/filament-excel#141] [Q&A]
Hi! I'm using Filament Table from a Livewire Project, and for some actions I am using simple buttons to create records instead of  built-in header actions, so for export I would like to execute it from a livewire button. How can I do it? 

al my livewire component, I did this function, but nothing happends:
```
    public function exportTable()
    {
        ExportAction::make('table')->exports([
            ExcelExport::make()
                ->fromModel(Product::class)
                ->withFilename(fn () => 'Export-' . date('Y-m-d'))
                ->withWriterType(\Maatwebsite\Excel\Excel::CSV)
                ->only([
                    'id', 'name',
                ])
        ]);

    }
```

**Comment by pxlrbt** [comment]
You could try using standalone Filament Actions in your component: https://filamentphp.com/docs/3.x/actions/installation

**Reply by swissjust-america** [reply]
nop, It shows the button but download does not start.

**Reply by pxlrbt** [reply]
Ah, nevermind. It's probably a different action class. Not sure how you would make that work. You could try to implement your own action that uses the ExportableAction trait.

**Reply by NekoMeowCat** [reply]
Good Day can anyone help me how can i use ->askForFileNames on a bulkaction on the export im using ur package sir pxlrbt

**Reply by pxlrbt** [reply]
Why don't you open your own discussion and provide some info and code instead of answering to a totally unreleated discussion?

**Comment by NekoMeowCat** [comment]
my sincere apologies. I posted a new discussion about it sir. Excuse my
ignorance it was my first time to reply on a thread on github I didn't mean
to come rude.

**Reply by pxlrbt** [reply]
No hard feelings. Just makes it easier to track the conversations
